release-team/hack/sync-bug-triage-github-project-beta.sh

```diff
@@ -0,0 +1,228 @@
+                        id
+                        fields(first:100) {
+                            nodes {
+                            ... on ProjectV2Field {
+                                id
+                                name
+                            }
+                            ... on ProjectV2SingleSelectField {
+                                id
+                                name
+                                options {
+                                    id
+                                    name
+                                }
+                            }
+                            }
+                        }
+                        }
+                    }
+                    }' -f org=${ORGANIZATION} -F number=${PROJECT_NUMBER} --jq "$query")"
+    echo $ID
+}
+
+
+function add_items_to_github_beta_project {
+    	    gh api graphql -f query='
+                    mutation (
+                        $project: ID!
+                        $item: ID!
+                        $type_field: ID!
+                        $option_id: String!
+                    ) {
+                        set_issue_type: updateProjectV2ItemFieldValue(input: {
+                            projectId: $project
+                            itemId: $item
+                            fieldId: $type_field
+                            value: {
+                                singleSelectOptionId: $option_id
+                            }
+                        }) {
+                            projectV2Item {
+                                id
+                            }
+                        }
+                    }' -f project="${PROJECT_ID}" -f item="${1}"  -f type_field="${TYPE_FIELD_ID}" -f option_id="${2}" --silent
+}
+
+echo 'Starting sync...'
+echo -e "[INFO] Fetching the list of open issues and PRs from k/k under the current release milestone, ${MILESTONE}"
+
+## Fetch issues
+MILESTONE_ISSUES_JSON="$(gh api graphql --paginate -f query='
+            query($org: String!, $repo: String!, $milestone: String!){
+              repository(owner: $org, name: $repo) {
+                description
+                url
+                milestones(states: [OPEN],first:1, query: $milestone) {
+                  nodes{
+                    issues(states:[OPEN], first: 200){
+                      nodes{
+                        id
+                      }
+                      pageInfo{
+                        hasNextPage
+                        endCursor
+                      }
+                    }
+                  }
+                  pageInfo{
+                    hasNextPage
+                    endCursor
+                  }
+                }
+              }
+            }' -f org=${ORGANIZATION} -f repo=${REPOSITORY} -f milestone="${MILESTONE}")"
+
+no_of_issues=$(jq ".data.repository.milestones.nodes[].issues.nodes | length" <<< "${MILESTONE_ISSUES_JSON}")
+issues_loop_index_range=$(( no_of_issues - 1 ))
```

Comment:
In the case of pagination, `no_of_issues` does not return a single number, but rather a number for each paginated group.

For example, I ran a modified query to force pagination:
```sh
MILESTONE_ISSUE_JSON="$(gh api graphql --paginate -f query='
            query($org: String!, $repo: String!, $milestone: String!, $endCursor: String){
              repository(owner: $org, name: $repo) {
                description
                url
                milestones(states: [OPEN],first:1, query: $milestone) {
                  nodes{
                    issues(states:[OPEN, CLOSED], first: 5, after: $endCursor){
                      nodes{
                        id
                      }
                      pageInfo{
                        hasNextPage
                        endCursor
                      }
                    }
                  }
                }
              }
            }' -f org=kubernetes -f repo=kubernetes -f milestone="v1.24")"
```

This resulted in the following prettified output:
```json
{
  "data": {
    "repository": {
      "description": "Production-Grade Container Scheduling and Management",
      "url": "https://github.com/kubernetes/kubernetes",
      "milestones": {
        "nodes": [
          {
            "issues": {
              "nodes": [
                {
                  "id": "MDU6SXNzdWUzNTc5MzcyNzc="
                },
                {
                  "id": "MDU6SXNzdWU3ODU0NjQwMTQ="
                },
                {
                  "id": "MDU6SXNzdWU4Mjk1OTg1NDM="
                },
                {
                  "id": "MDU6SXNzdWU4MzMwNzU2Mzc="
                },
                {
                  "id": "MDU6SXNzdWU5Mzc0MzIxNjg="
                }
              ],
              "pageInfo": {
                "hasNextPage": true,
                "endCursor": "Y3Vyc29yOnYyOpLAzjfgFGg="
              }
            }
          }
        ]
      }
    }
  }
}
{
  "data": {
    "repository": {
      "description": "Production-Grade Container Scheduling and Management",
      "url": "https://github.com/kubernetes/kubernetes",
      "milestones": {
        "nodes": [
          {
            "issues": {
              "nodes": [
                {
                  "id": "MDU6SXNzdWU5MzgyNzE1Nzc="
                },
                {
                  "id": "MDU6SXNzdWU5NjYxMDUzNDM="
                },
                {
                  "id": "MDU6SXNzdWU5ODE2OTg2MzE="
                },
                {
                  "id": "I_kwDOAToIks48R-_H"
                },
                {
                  "id": "I_kwDOAToIks485qjC"
                }
              ],
              "pageInfo": {
                "hasNextPage": true,
                "endCursor": "Y3Vyc29yOnYyOpLAzjzmqMI="
              }
            }
          }
        ]
      }
    }
  }
}
{
  "data": {
    "repository": {
      "description": "Production-Grade Container Scheduling and Management",
      "url": "https://github.com/kubernetes/kubernetes",
      "milestones": {
        "nodes": [
          {
            "issues": {
              "nodes": [
                {
                  "id": "I_kwDOAToIks49mPsO"
                },
                {
                  "id": "I_kwDOAToIks4_Jl9m"
                },
                {
                  "id": "I_kwDOAToIks5DVQcj"
                },
                {
                  "id": "I_kwDOAToIks5EccUj"
                },
                {
                  "id": "I_kwDOAToIks5F5hqn"
                }
              ],
              "pageInfo": {
                "hasNextPage": true,
                "endCursor": "Y3Vyc29yOnYyOpLAzkXmGqc="
              }
            }
          }
        ]
      }
    }
  }
}
{
  "data": {
    "repository": {
      "description": "Production-Grade Container Scheduling and Management",
      "url": "https://github.com/kubernetes/kubernetes",
      "milestones": {
        "nodes": [
          {
            "issues": {
              "nodes": [
                {
                  "id": "I_kwDOAToIks5GL4xf"
                },
                {
                  "id": "I_kwDOAToIks5GMppR"
                },
                {
                  "id": "I_kwDOAToIks5GNW7c"
                },
                {
                  "id": "I_kwDOAToIks5GQSss"
                },
                {
                  "id": "I_kwDOAToIks5GRJuM"
                }
              ],
              "pageInfo": {
                "hasNextPage": true,
                "endCursor": "Y3Vyc29yOnYyOpLAzkZEm4w="
              }
            }
          }
        ]
      }
    }
  }
}
{
  "data": {
    "repository": {
      "description": "Production-Grade Container Scheduling and Management",
      "url": "https://github.com/kubernetes/kubernetes",
      "milestones": {
        "nodes": [
          {
            "issues": {
              "nodes": [
                {
                  "id": "I_kwDOAToIks5GRkQm"
                },
                {
                  "id": "I_kwDOAToIks5GZYK2"
                },
                {
                  "id": "I_kwDOAToIks5GaF2M"
                },
                {
                  "id": "I_kwDOAToIks5GkPkh"
                },
                {
                  "id": "I_kwDOAToIks5GmTJS"
                }
              ],
              "pageInfo": {
                "hasNextPage": true,
                "endCursor": "Y3Vyc29yOnYyOpLAzkaZMlI="
              }
            }
          }
        ]
      }
    }
  }
}
{
  "data": {
    "repository": {
      "description": "Production-Grade Container Scheduling and Management",
      "url": "https://github.com/kubernetes/kubernetes",
      "milestones": {
        "nodes": [
          {
            "issues": {
              "nodes": [
                {
                  "id": "I_kwDOAToIks5Gtduc"
                },
                {
                  "id": "I_kwDOAToIks5Gvivj"
                },
                {
                  "id": "I_kwDOAToIks5GxtNC"
                },
                {
                  "id": "I_kwDOAToIks5G48Xk"
                },
                {
                  "id": "I_kwDOAToIks5G-4-C"
                }
              ],
              "pageInfo": {
                "hasNextPage": true,
                "endCursor": "Y3Vyc29yOnYyOpLAzkb7j4I="
              }
            }
          }
        ]
      }
    }
  }
}
{
  "data": {
    "repository": {
      "description": "Production-Grade Container Scheduling and Management",
      "url": "https://github.com/kubernetes/kubernetes",
      "milestones": {
        "nodes": [
          {
            "issues": {
              "nodes": [
                {
                  "id": "I_kwDOAToIks5HBDuK"
                },
                {
                  "id": "I_kwDOAToIks5IhDUu"
                }
              ],
              "pageInfo": {
                "hasNextPage": false,
                "endCursor": "Y3Vyc29yOnYyOpLAzkiENS4="
              }
            }
          }
        ]
      }
    }
  }
}
```

Which when filtered through jq for length results in:

```sh
> echo "${MILESTONE_ISSUE_JSON}" | jq ".data.repository.milestones.nodes[].issues.nodes | length"
5
5
5
5
5
5
2
```

I'm not sure if there is a way within jq to produce the proper length, but you could always wrap this with an outer loop that consumes each line of the length output separately

Reply:
Thank you so much for including the example above, @detiber 

(it actually forced me to try the "forced pagination" myself, and only then was I able to understand the issue and proceed.) 🙂 

I've addressed the changes introduced by the pagination in the script.